manifests/forwarder/config.pp

```diff
@@ -0,0 +1,110 @@
+
+  if $splunk::forwarder::manage_password {
+    file { $splunk::forwarder::password_config_file:
+      ensure  => file,
+      owner   => $splunk::forwarder::splunk_user,
+      group   => $splunk::forwarder::splunk_user,
+      content => $splunk::forwarder::password_content,
+    }
+
+    file { $splunk::forwarder::secret_file:
+      ensure  => file,
+      owner   => $splunk::forwarder::splunk_user,
+      group   => $splunk::forwarder::splunk_user,
+      content => $splunk::forwarder::secret,
+    }
+  }
+
+  # Remove init.d file if the service provider is systemd
+  if $facts['service_provider'] == 'systemd' and versioncmp($splunk::forwarder::version, '7.2.2') >= 0 {
+    file { '/etc/init.d/splunk':
+      ensure => 'absent',
+    }
+  }
+
+
+  $_forwarder_file_mode = $facts['kernel'] ? {
+    'windows' => undef,
+    default   => '0600',
+  }
+
+  file { ["${splunk::forwarder::forwarder_homedir}/etc/system/local/deploymentclient.conf",
+          "${splunk::forwarder::forwarder_homedir}/etc/system/local/outputs.conf",
+          "${splunk::forwarder::forwarder_homedir}/etc/system/local/inputs.conf",
+          "${splunk::forwarder::forwarder_homedir}/etc/system/local/props.conf",
+          "${splunk::forwarder::forwarder_homedir}/etc/system/local/transforms.conf",
+          "${splunk::forwarder::forwarder_homedir}/etc/system/local/web.conf",
+          "${splunk::forwarder::forwarder_homedir}/etc/system/local/limits.conf",
+          "${splunk::forwarder::forwarder_homedir}/etc/system/local/server.conf"]:
+    ensure => file,
+    tag    => 'splunk_forwarder',
+    owner  => $splunk::forwarder::splunk_user,
+    group  => $splunk::forwarder::splunk_user,
+    mode   => $_forwarder_file_mode,
+  }
+
+  if $splunk::forwarder::use_default_config {
+    splunkforwarder_web { 'forwarder_splunkd_port':
+      section => 'settings',
+      setting => 'mgmtHostPort',
+      value   => "${splunk::forwarder::splunkd_listen}:${splunk::forwarder::splunkd_port}",
+      tag     => 'splunk_forwarder',
+    }
+
+    $splunk::forwarder::forwarder_input.each | String $name, Hash $options| {
+      splunkforwarder_input { $name:
+        section => $options['section'],
+        setting => $options['setting'],
+        value   => $options['value'],
+        tag     => 'splunk_forwarder',
+      }
+    }
+    $splunk::forwarder::forwarder_output.each | String $name, Hash $options| {
+      splunkforwarder_output { $name:
+        section => $options['section'],
+        setting => $options['setting'],
+        value   => $options['value'],
+        tag     => 'splunk_forwarder',
+      }
+    }
+  }
+
+  # Declare addons
+  create_resources('splunk::addon', $splunk::forwarder::addons)
+
+  File <| tag == 'splunk_forwarder' |>
+  -> Splunkforwarder_deploymentclient <| tag == 'splunk_forwarder' |>
+  ~> Class['splunk::forwarder::service']
+
+  File <| tag == 'splunk_forwarder' |>
+  -> Splunkforwarder_input            <| tag == 'splunk_forwarder' |>
```

Comment:
What's the deal with only looking for resources with the specific tag?
Previously, you could declare these resources without a tag and they would notify the service.
See https://github.com/voxpupuli/puppet-splunk/pull/50